### PR TITLE
Encapsulate snapshot configuration logic

### PR DESCRIPTION
**Pull request by morgando**
The changes in this PR encapsulate logic for configuring snapshot.

### CODE REVIEW

File: db/config.c [hunk 1]
```diff
@@ -684,6 +685,85 @@ static int new_table_from_schema(struct dbenv *dbenv, char *tblname,
...
+    if (gbl_snap_impl == impl) {
+        return;
+    }
+
+    gbl_snap_impl = impl;
+    gbl_snap_impl == SNAP_IMPL_MODSNAP ? toggle_modsnap(1) : toggle_modsnap(0);
+    gbl_snap_impl == SNAP_IMPL_NEW ? toggle_new_snapisol(1) : toggle_new_snapisol(0);
+
+    logmsg(LOGMSG_USER, "%s: Switched snapshot implementation to '%s'\n",
+                        __func__, snap_impl_str(gbl_snap_impl));
```
Comment (morgando) [inline db/config.c:747]: Is this log-level too chatty?

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Error &#x26A0;.<br>Regression testing: <b>1/402 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_truncate</p>

File: db/comdb2.c [hunk 2]
```diff
@@ -459,6 +459,8 @@ int gbl_enable_berkdb_retry_deadlock_bias = 0;
 int gbl_enable_cache_internal_nodes = 1;
 int gbl_use_modsnap_for_snapshot = 0;
 int gbl_modsnap_asof = 0;
+const int gbl_snap_fallback_impl = SNAP_IMPL_ORIG;
+int gbl_snap_impl = gbl_snap_fallback_impl;;
```
Comment (chands10) [inline db/comdb2.c:463]: Double semicolon

File: db/config.c (hunk 1 above)
Comment (chands10) [reply db/config.c:747]: Seems fine I think, wouldn't expect to be switching that often?

**Review by chands10 — DISMISSED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>2/402 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>guid<br>truncatesc</p>

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>0/0 tests failed</b> &#x26A0;.</p>

**Review by chands10 — APPROVED** [review]

**Review by roborivers — CHANGES_REQUESTED** [review]
<p><a href='https://bloomberg.github.io/comdb2/contrib.html#code-style'>Coding style</a> check: <b>Error</b>. &#x26A0;.<br>Smoke testing: Success &#10003;.<br>Cbuild submission: Success &#10003;.<br>Regression testing: <b>4/562 tests failed</b> &#x26A0;.</p><p>The first 10 failing tests are: <br>sc_resume<br>insert_lots_large_tran_generated<br>online_compaction<br>sys_cmd_access</p>

**Review by chands10 — APPROVED** [review]

**Comment by morgando** [comment]
`mdouglas47 Jul 30 13:17:58 2024  in-comdb2db refactor_snapshot_config.R20240730.9`

**Pull request merged**